app/src/main/java/com/tari/android/wallet/ui/activity/tx/TxDetailsActivity.kt

```diff
@@ -135,18 +143,30 @@ internal class TxDetailsActivity : AppCompatActivity(), ServiceConnection {
         ui = ActivityTxDetailsBinding.inflate(layoutInflater).apply { setContentView(root) }
         overridePendingTransition(R.anim.enter_from_right, R.anim.exit_to_left)
         setupUI()
-        tracker.screen(path = "/home/tx_details", title = "Transaction Details")
-    }
 
-    override fun onStart() {
-        super.onStart()
-        // start service if not started yet
+        val savedTx = savedInstanceState?.getParcelable<Tx>(TX_EXTRA_KEY)
+        val intentTx = intent.getParcelableExtra<Tx>(TX_EXTRA_KEY)
+        when {
+            savedTx != null -> {
+                tx = savedTx
+                bindTxData()
+                observeTxUpdates()
+                enableCTAs()
+            }
+            intentTx != null -> {
+                tx = intentTx
+                bindTxData()
+                observeTxUpdates()
+                enableCTAs()
+            }
+        }
         if (walletService == null) {
             // bind to service
             val bindIntent = Intent(this, WalletService::class.java)
             Logger.d("Issuing bindService (${System.currentTimeMillis()})")
             bindService(bindIntent, this, Context.BIND_AUTO_CREATE)
         }
+        tracker.screen(path = "/home/tx_details", title = "Transaction Details")
```

Comment:
won't it be called on every config change?

Reply:
correct, fixing.